packages/immutable-arraybuffer/src/limited-pony-for-hermes.js

```diff
@@ -95,10 +75,24 @@ ImmutableArrayBufferInternal.prototype = {
     throw TypeError('Cannot resize an immutable ArrayBuffer');
   },
 };
-// Since it should not escape, this `freeze` is just belt-and-suspenders.
-// `freeze` is shallow, so `ImmutableArrayBufferInternal.prototype` remains
-// unfrozen.
-Object.freeze(ImmutableArrayBufferInternal);
+
+/**
+ * Emulates the encapsulated `ImmutableArrayBufferInternal` class constructor
+ * from the original except this function takes the `realBuffer` which its
+ * result encapsulates. Security demands that this result has exclusive access
+ * to the `realBuffer` it is given, which its callers must ensure.
+ *
+ * @param {ArrayBuffer} realBuffer
+ * @returns {ArrayBuffer}
+ */
+const makeImmutableArrayBufferInternal = realBuffer => {
```

Comment:
Since this instance creator is purely internal, there was no need for it to emulate a class. I just made it an array function that doesn't mention `this`. As a result, I was able to figure out how to fix the types.